presto-main/src/main/java/io/prestosql/execution/CreateTableTask.java

```diff
@@ -161,6 +162,10 @@ else if (element instanceof LikeClause) {
 
                 TableMetadata likeTableMetadata = metadata.getTableMetadata(session, likeTable);
 
+                accessControl.checkCanShowColumns(
+                        session.toSecurityContext(),
+                        new CatalogSchemaTableName(likeTableName.getCatalogName(), likeTableName.asSchemaTableName()));
+
```

Comment:
`checkCanShowColumns` is dedicated for `SHOW COLUMNS` statement. Notice that raising error: `Cannot show columns of table %s` for `CREATE TABLE` could be a bit misleading. 

Instead I would use `io.prestosql.security.AccessControl#filterColumns`, then in case all columns were filtered you might need to raise some custom `AccessDeniedException`

Reply:
If we model after DESCRIBE then it should work like that:
1.checkCanShowColumns
2.filterColumns and use only those that are visible
WDYT?

Reply:
Do you `io.prestosql.sql.rewrite.ShowQueriesRewrite.Visitor#visitShowColumns` and `filterColumns` in `io.prestosql.metadata.MetadataListing#listTableColumns`?

That sounds good to me, but I think we might need to have some generic `AccessControl::checkCanReferenceTable` instead of using `visitShowColumns` here.

@electrum, @martint WDYT?

Reply:
What would be the semantics of `checkCanReferenceTable`? That seems to overlap with checking for SELECT permissions, so it'd be error prone if the implementations of both methods are inconsistent with each other.

Reply:
> What would be the semantics of checkCanReferenceTable? 

IMO it should allow to reference table anytime it is not filtered by `filterTables`. So if user has any privilege to a table or any its columns.

As I understand user does not have to have SELECT access to do CREATE TABLE LIKE. For example INSERT could also be good enough here.

Reply:
One more though. Since we are requiring admin-like privileges for SHOW CREATE TABLE as that could reveal sensitive data (table properties) then maybe we need admin-like privileges here as well. Because here we also could access sensitive data. It is a bit unitive to me, but it is coherent and secure. That way we could simply call `checkCanShowCreateTable` here and overwrite the error message that table cannot be used in `CREATE TABLE LIKE`.

@skrzypo987 @martint What do you think? Then this pull request could become very simple. No need to do `filterColumns` etc.

Reply:
This seems reasonable If and only if the LIKE statement can retrieve data that DESCRIBE cannot.

Reply:
> This seems reasonable If and only if the LIKE statement can retrieve data that DESCRIBE cannot.

So this is the case when we are copying table properties. If we are just copying columns then maybe describe is enough, but what about column filtering, it seems needed then.